src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,127 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private int size;
+    private Object[] currentList;
+    private Object[] newList;
+
+    public ArrayList() {
+        size = 0;
+        currentList = new Object[DEFAULT_CAPACITY];
+        newList = new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == currentList.length) {
+            grow();
+            System.arraycopy(currentList, 0, newList, 0, size);
+            currentList = newList;
+        }
+        currentList[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("Index is out of bounds");
+        }
```

Comment:
```suggestion
        if (index < 0 || index > size) {
            throw new ArrayListIndexOutOfBoundsException("Index is out of bounds");
        }
```

move to the separate method